### ISSUE TITLE
[Bug]: Problems building with 'ccmake ..' on Linux Mint 21.1

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [brndnmtthws/conky#1348]
### What happened?

I tried to compile Conky 1.16.1 from source, using `ccmake ..`, on Linux Mint 21.1 Cinnamon. Some build problems I solved, but I am stuck on a problem that I cannot solve. _The building problems that I **did** solve_, I solved thusly:

1) `$ export TERM=xterm`
2) `$ sudo ln -s /usr/bin/cmake /usr/local/bin/cmake`

The _persisting problem_ is as follows.

3) `ccmake ..` yields:

```
 CMake Error at cmake/ConkyPlatformChecks.cmake:335 (message):
   Unable to find freetype library
 Call Stack (most recent call first):
   CMakeLists.txt:35 (include)
```

My unsuccessful efforts to solve problem _3_ include these: using `cmake-gui` to add various entries pointing to locations to do with freetype; reinstalling various various freetype packages.

The same three problems occurred (or was it just _1_ and _2_?) when I tried to build Conky versions 1.16 and 1.15 - in both cases using `ccmake ..`.

Building Conky 1.16.1 via `cmake ..` as against `ccmake ..`, i.e. with default configuration options, works - except that, since that configuration excludes wifi support, my conky script has trouble! To wit:

```
conky: unknown variable '$wireless_link_qual_perc'
conky: Bad arguments: '${wireless_link_qual_perc}' and '69'
conky: compare failed for expression '${wireless_link_qual_perc}>69'
```

I am sorry to bother you at Christmas, but I see that already, today, you pushed out a new version (namely, version 1.16.1).

### Version

1.16.1

### Which OS/distro are you seeing the problem on?

Linux (other)

**Comment by brndnmtthws** [comment]
Are you by any chance using any previously cached values? i.e., are you running cmake with a `CMakeCache.txt` file in the build dir?

**Comment by LinuxOnTheDesktop** [comment]
No, II'm afraid not. Even running ``ccmake ..` with an empty 'build' directory produces the freetype error.

**Comment by LinuxOnTheDesktop** [comment]
I find that `CMakeError.log` contains (among other stuff) the following, at its end - and I _think_ this material has to do with the latest of my failed attempts to build.

```
Determining if the function iw_sockets_open exists failed with the following output:
Change Dir: /home/[my user name]/git/conky/conky-1.16.1/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_f32f7/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_f32f7.dir/build.make CMakeFiles/cmTC_f32f7.dir/build
gmake[1]: Entering directory '/home/[my user name]/git/conky/conky-1.16.1/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_f32f7.dir/CheckFunctionExists.c.o
/usr/bin/cc -D_GNU_SOURCE  -DCHECK_FUNCTION_EXISTS=iw_sockets_open   -o CMakeFiles/cmTC_f32f7.dir/CheckFunctionExists.c.o   -c /usr/local/share/cmake-3.11/Modules/CheckFunctionExists.c
Linking C executable cmTC_f32f7
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_f32f7.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=iw_sockets_open    CMakeFiles/cmTC_f32f7.dir/CheckFunctionExists.c.o  -o cmTC_f32f7 
/usr/bin/ld: CMakeFiles/cmTC_f32f7.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x14): undefined reference to `iw_sockets_open'
collect2: error: ld returned 1 exit status
gmake[1]: *** [CMakeFiles/cmTC_f32f7.dir/build.make:87: cmTC_f32f7] Error 1
gmake[1]: Leaving directory '/home/[my user name]/git/conky/conky-1.16.1/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:126: cmTC_f32f7/fast] Error 2
```

**Comment by LinuxOnTheDesktop** [comment]
Yesterday I tried to build from the latest git (which contains commits that have yet to be released; the last commit that I have is #31d247a80d11441f7d4fb4416ea885f22aed66dd; so I think that what I write here will apply, also, to the very newly released version 1.17). Using that code, I no longer get the freetype error (which is the error that I mention above). Yet, I do - while trying to configure the build options, via `cmake ..` - encounter the following, fatal, error message.

> Unknown CMake command "add_link_options"

Searching on the web suggests that the problem owes to having a version of cmake earlier than 3.12. But I have cmake 3.22.1 (_sic_).

**Comment by brndnmtthws** [comment]
`add_link_options` is new in 3.13, according to https://cmake.org/cmake/help/latest/command/add_link_options.html.

However, I believe the issue is actually because of some generator expression features that were recently added: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

Specifically, `COMPILE_LANG_AND_ID`, but that should be available in >=3.15. I may have included one of the newer ones and forgot about it, a bunch were added in cmake 3.24.

**Comment by LinuxOnTheDesktop** [comment]
Thanks.

I'd rather not try to update my cmake (from 3.22.1 to >= 3.24) in case it causes problems on my computers. So, please advise. I imagine that quite a few people who use Ubuntu and Ubuntu-derivatives such as Mint and Pop_OS are going to have the problem that I am having (at least if they want to configure the build of Conky that they get).

**Comment by brndnmtthws** [comment]
I tested with cmake 3.22.1, and it built fine for me. Are you sure you are running the cmake binaries for 3.22.1?

**Comment by LinuxOnTheDesktop** [comment]
> Are you sure you are running the cmake binaries for 3.22.1?

Well:

```
$ which cmake
/usr/bin/cmake
$ cmake version
cmake version 3.22.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

**Comment by LinuxOnTheDesktop** [comment]
Seemingly there is indeed something wrong at my end. For, I have two ThinkPads, running the same OS and indeed the two PCs are very similarly configured. On one of the PCs, I have the problem that I am describing. On the other, I find, I could build and configure Conky 1.17 without problems. So I built a .deb from the successful installation and used it to install Conky on the other PC. So: problem sort of solved?

**Comment by LinuxOnTheDesktop** [comment]
I got cmake (`cmake ..`) working on the PC on which it was not working. _I did so by using a newer form of cmake_. Let me now spell out that procedure.

I downloaded cmake 3.24.3, in the form of cmake-3.24.3-linux-x86_64.sh, from  [here](https://cmake.org/download/). (That file somehow wraps cmake into a shell script!)

I used the file to install cmake - together with ccmake, and friends - into a user directory.

When building Conky, I did not run `cmake ..` - which would invoke my system-installed cmake. Rather I ran `path-to-my-local-cake/cmake ..`.

**Comment by brndnmtthws** [comment]
Cool, sounds like we can close this now :)

**Comment by Lssg97** [comment]
Hi guys,

Sorry for opening this issue again, but the information I searched on the Internet is too little to solve this problem independently.

I also got the "Unable to find freetype library" error. I'm using a 2018 MacBook Air with Mojave (10.14.6). All dependencies have been installed according to the method introduced in the wiki. But this error still occurs when executing `ccmake ..`.

The following shows the screenshots of the errors I encountered, as well as the versions of the dependent software:
<img width="1082" alt="屏幕快照 2023-07-13 22 56 15" src="https://github.com/brndnmtthws/conky/assets/47817993/639bfa14-335f-4b6e-aad1-4028c1f59ec0">
<img width="1082" alt="屏幕快照 2023-07-13 23 01 24" src="https://github.com/brndnmtthws/conky/assets/47817993/1eda2746-4e6e-4bf2-808d-32e07b9998da">
PS: The information I found on the Internet is that this library is related to PHP, but there is freetype in the installed dependencies, so I guess these messages may have nothing to do with the problems I encountered.
